### ISSUE TITLE
Questions regarding TF2.4 OD Model Zoo

### ISSUE THREAD

**Issue by WonsokYoo** [openvinotoolkit/openvino#4952]
##### System information (version)
- OpenVINO=> #4772
- Operating System / Platform => Ubuntu18.04
- Problem classification => Model Conversion
- Model name: (if applicable)
- -->centernet_resnet50_v2_512x512_coco17 (TF2 Model Zoo)
##### Detailed description
@lazarevevgeny Hello again, thanks for your support I'm working on OpenVINO very well.
This time I tried 'centernet_resnet50_v2_512x512_coco17' but failed to MO.
I've read your past #3556 PR which is merged into master branch but this seems like not supporting 'TF2.4' as well as 'CenterNet' topologies.
I wonder if you are considering about 'CenterNet' in your new PR #4772 .

**Comment by lazarevevgeny** [comment]
@WonsokYoo , unfortunately there are some unsupported operations which prevents us from enabling CenterNet family of models. But I will keep looking into this direction in the future.

**Comment by WonsokYoo** [comment]
@lazarevevgeny Thanks for your reply, I hope the enhancement will come soon. I will keep tracking this repository.

**Comment by b04505009** [comment]
From [tensorflow detection model zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md) , CenterNet MobileNetV2 FPN 512x512 got very fast speed (6 ms) and acceptable accuracy (23.4 mAP). Hope OpenVINO can support this powerful model soon!

**Comment by lazarevevgeny** [comment]
@b04505009 , thank you for keep interesting in OpenVINO. We were working hard to enable some additional operations to support these models, but unfortunately not all of them will be done in upcoming release. 

However, if you are comfortable with building OpenVINO from sources you can take the MO from this PR: https://github.com/openvinotoolkit/openvino/pull/5438/ and convert the model without use of json configuration file. Just specify the `--input_shape [1,<HEIGHT>,<WIDTH>,3]` command line parameter. 

Then take Inference Engine from this PR https://github.com/openvinotoolkit/openvino/pull/5169 and compile it. Then you will be able to infer the CenterNet model using IE compiled from sources.

**Comment by b04505009** [comment]
@lazarevevgeny 
Thank you for your detailed information! I will try it later.

**Comment by b04505009** [comment]
@lazarevevgeny 
I noticed that #5169 cannot directly patch on master branch, so should I download directly from https://github.com/evolosen/openvino/tree/ngraph/if_prototype or there are somethings that I have to do before patch on the master branch?
Thank you

**Comment by lazarevevgeny** [comment]
@b04505009 , please, download the MO from this branch.

**Comment by b04505009** [comment]
@lazarevevgeny 
I tried to convert centernet_mobilenetv2_fpn to IR from openvino built from this PR #5438, but some errors occur:
```
[ ERROR ]  Cannot infer shapes or values for node "StatefulPartitionedCall/map/TensorArrayV2_2".
[ ERROR ]  Tensorflow type 21 not convertible to numpy dtype.
[ ERROR ]  
[ ERROR ]  It can happen due to bug in custom shape infer function <function tf_native_tf_node_infer at 0x7f3265a6d5f0>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the  or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.
[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "StatefulPartitionedCall/map/TensorArrayV2_2" node. 
```

Could you help me to figure out the problem? Thank you
Here is my [model](https://drive.google.com/file/d/1Xm4FPzH-qx2waRan2o6frEKCcKLYHlaf/view?usp=sharing)

My command:
```
python openvino_centernet_mo/deployment_tools/model_optimizer/mo_tf.py \
    --saved_model_dir training/centernet_mobilenetv2_fpn_0521/exported/saved_model \
    --input_shape "[1,512,512,3]" \
    --output_dir training/centernet_mobilenetv2_fpn_0521/IR_model/FP32 \
    --data_type FP32 \
    --tensorflow_object_detection_api_pipeline_config training/centernet_mobilenetv2_fpn_0521/centernet_mobilenetv2_fpn_0521.config \
    --reverse_input_channels
```

**Comment by lazarevevgeny** [comment]
@evolosen , could you help @b04505009 to convert the CenterNet model from your branch?

**Comment by b04505009** [comment]
Sorry, is there any progress? Thank you so much.

**Comment by evolosen** [comment]
Unfortunately, MO cant convert model centernet_resnet50_v2_512x512_coco17. There are operations which we do not support yet. You can use another "CenterNet" model (for example 'centernet_resnet101_v1_fpn_512x512_coco17_tpu-8') which does not contains unsupported operations.

**Comment by evolosen** [comment]
Unfortunately, MO cant convert model CenterNet 'centernet_mobilenetv2fpn_512x512_coco17'. There are operations which we do not support yet. You can use another "CenterNet" model (for example 'centernet_resnet101_v1_fpn_512x512_coco17_tpu-8') which does not contains unsupported operations.

**Comment by evolosen** [comment]
@b04505009, I'm sorry, My answer was incorrect. MO with PR [#5438](https://github.com/openvinotoolkit/openvino/pull/5438) can convert CenterNet 'centernet_mobilenetv2fpn_512x512_coco17' from [mo  tensorflow detection model zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md). But your [model ](https://drive.google.com/file/d/1Xm4FPzH-qx2waRan2o6frEKCcKLYHlaf/view) contains unsupported  operations.

**Comment by b04505009** [comment]
@evolosen However, this thing is weird because my config [centernet_mobilenetv2_fpn_0521.config.zip](https://github.com/openvinotoolkit/openvino/files/6600009/centernet_mobilenetv2_fpn_0521.config.zip) only change the num_classes, batch_size, and data settings from the official one [centernet_mobilenetv2_fpn_od.zip](https://github.com/openvinotoolkit/openvino/files/6600015/centernet_mobilenetv2_fpn_od.zip). It should not contain additional operations. Do you have some idea about this? Thank you for your help.

**Comment by evolosen** [comment]
b04505009б I think the trouble in your version of 'centernet_mobilenetv2fpn_512x512_coco17' model. I know direct link model([model zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md) ->[CenterNet MobileNetV2 FPN 512x512](http://download.tensorflow.org/models/object_detection/tf2/20210210/centernet_mobilenetv2fpn_512x512_coco17_od.tar.gz)) will be converted. 
You may be using a newer model that has operations that the MO does not support yet. Perhaps you should find an earlier version of this model that does not contain these operations and this problem.

**Comment by b04505009** [comment]
@evolosen I found that I can't even convert the official pretrained model. Here's the error message:
```
[ ERROR ]  Shape [ 1 -1 -1 64] is not fully defined for output 0 of "StatefulPartitionedCall/center_net_mobile_net_v2fpn_feature_extractor/model_1/tf.__operators__.add/AddV2". Use --input_shape with positive integers to override model input shapes.
[ ERROR ]  Cannot infer shapes or values for node "StatefulPartitionedCall/center_net_mobile_net_v2fpn_feature_extractor/model_1/tf.__operators__.add/AddV2".
[ ERROR ]  Not all output shapes were inferred or fully defined for node "StatefulPartitionedCall/center_net_mobile_net_v2fpn_feature_extractor/model_1/tf.__operators__.add/AddV2". 
 For more information please refer to Model Optimizer FAQ, question #40. (https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=40#question-40)
[ ERROR ]  
[ ERROR ]  It can happen due to bug in custom shape infer function <function Elementwise.__init__.<locals>.<lambda> at 0x7fe9416584d0>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.
[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "StatefulPartitionedCall/center_net_mobile_net_v2fpn_feature_extractor/model_1/tf.__operators__.add/AddV2" node. 
 For more information please refer to Model Optimizer FAQ, question #38. (https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=38#question-38)
```
My command is like:
```
source ./openvino_centernet_mo/bin/setupvars.sh
python openvino_centernet_mo/deployment_tools/model_optimizer/mo_tf.py \
    --saved_model_dir training/centernet_mobilenetv2_fpn_od/saved_model \
    --input_shape "[1,512,512,3]" \
    --output_dir training/centernet_mobilenetv2_fpn_od/IR_model/FP32 \
    --data_type FP32 \
    --tensorflow_object_detection_api_pipeline_config training/centernet_mobilenetv2_fpn_od/pipeline.config
```
How I clone your PR:
```
git clone https://github.com/openvinotoolkit/openvino openvino_centernet_mo_src
cd openvino_centernet_mo_src
git fetch origin pull/5438/head:centernet_mo
git checkout centernet_mo
git submodule update --init --recursive
mkdir build
ccmake ..
cd build
ccmake ..
make -j8
DESTDIR=../../openvino_centernet_mo make install
```
Could you help me to check which step I've made some mistakes in? Thank you so much

**Comment by evolosen** [comment]
b04505009 , I am sorry for a long answer. The trouble in incorrect input shape. Try it --input_shape "[1,320,320,3]". As I think the  model expects the shape "[1,320,320,3]".

**Comment by evolosen** [comment]
b04505009 any other questions?

**Comment by b04505009** [comment]
Hi @evolosen, sorry for the late reply. It is weird that the input_shape is 320x320 while the dimension in the config is 512x512. There may be some mismatch between models and config, but it's tensorflow's issue. For OpenVINO I don't have further questions. Thanks again for your help.
